app/models/rate_limited_search.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class RateLimitedSearch < Common::RedisStore
+  COUNT_LIMIT = 3
+
+  redis_store(name.underscore)
+  redis_ttl(86_400)
```

Comment:
Looks like 'expire' is set to this amount every save.
So, if someone tries 3 times, then tries again every 23 hours after that, they are going to have a bad time.
Probably not a practical issue, but thought I'd mention it.